### PR TITLE
OpenStack: Remove cloud-creds-secret-reader rolebinding

### PR DESCRIPTION
**Pull request by Fedosin**
OpenStack deployments don't need this rolebinding anymore, so we can remove it now.

Initially it was used to allow kubelet to obtain cloud provider credentials from the secret. Later we rearchitect the provider, and now it doesn't need this access, because it is able to read required data from the metadata service, like other providers do.

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
Can you provide details around how it was used before, and what changes make it not required anymore?

**Comment by Fedosin** [comment]
@abhinavdahiya I updated the description

File: pkg/asset/manifests/openshift.go [hunk 1]
```diff
@@ -177,11 +177,6 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 		assetData["99_role-cloud-creds-secret-reader.yaml"] = applyTemplateData(roleCloudCredsSecretReader.Files()[0].Data, templateData)
 	}
 
-	switch platform {
-	case openstacktypes.Name:
-		assetData["99_rolebinding-cloud-creds-secret-reader.yaml"] = applyTemplateData(roleBindingCloudCredsSecretReader.Files()[0].Data, templateData)
```
Comment (abhinavdahiya) [inline pkg/asset/manifests/openshift.go]: it looks like this was the only consumer of asset `roleBindingCloudCredsSecretReader` can we remove `roleBindingCloudCredsSecretReader` too?

Comment (Fedosin) [reply pkg/asset/manifests/openshift.go]: sure, will be done!

**Comment by Fedosin** [comment]
/hold
I'm going to hold it for a while until OpenStack CI works again

**Pull request merged**